agcaballero/walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -35,144 +46,130 @@
     Collection<String> attendees = request.getAttendees();
     Collection<String> optionalAttendees = request.getOptionalAttendees();
     
-    // are there any mandatory employees?
-    boolean mandatory = false;
-    if(attendees.size() > 0) {
-      mandatory = true;
-    }
-    
-    // are there any optional employees?
-    boolean optional = false; 
-    if(optionalAttendees.size() > 0) {
-      optional = true;
-    }
+    // Holds whether the mandatory attendees can attend & the number of optional attendees 
+    // that can attend. If minuteStatus[number] is equal to MANDATORY_UNAVAILABLE, then mandatory
+    // attendees can't attend at that number. If it's equal to 0, then all attendees can attend.
+    // If it's equal to a negative int, then the abs value of that negative number is the number 
+    // of optional attendees who can't attend. The +1 is to account for last minute in day
+    int[] minuteStatus = new int[MINUTES_IN_DAY + 1];
```

Comment:
This type of system is hard to understand/maintain. Is it possible to use a class instead?

```
class Availability {
  boolean areMandatoryAttendeesAllAvailable;
  int numberOfOptionalAttendeesAvailable;
}
```

Reply:
Yes. I will make the change.

Reply:
Done